avm/src/vm/program.rs

```diff
@@ -164,6 +167,33 @@ impl Program {
       .get_mut(&cur_handler.event_id)
       .unwrap();
     handlers.push(cur_handler);
+    // special logic to auto-listen on the http server if it is being used
+    if program.event_handlers.get(&BuiltInEvents::HTTPCONN.into()).unwrap().len() > 0 {
+      let start_handlers = program.event_handlers.get_mut(&BuiltInEvents::START.into()).unwrap();
+      if start_handlers.len() == 0 {
+        // Create a new handler that just executes `httplsn`
+        let mut listen_handler = EventHandler::new(1, BuiltInEvents::START.into());
+        listen_handler.add_instruction(Instruction {
+          id: 0,
+          opcode: OPCODES.get(&opcode_id("httplsn")).unwrap(),
+          args: vec![0, 0, 0],
+          dep_ids: vec![],
+        });
+        start_handlers.push(listen_handler);
+      } else {
+        // Append the listen opcode to the end of the first start handler found
+        let start_handler = &mut start_handlers[0];
+        let start_handler_fragment_idx = start_handler.fragments.len() - 1;
+        let last_frag = &mut start_handler.fragments[start_handler_fragment_idx];
+        let last_id = last_frag[last_frag.len() - 1].id;
+        start_handler.add_instruction(Instruction {
+          id: i64::MAX, // That shouldn't collide with anything
+          opcode: OPCODES.get(&opcode_id("httplsn")).unwrap(),
+          args: vec![0, 0, 0],
+          dep_ids: vec![last_id], // To make sure this is the last instruction run (usually)
+        });
```

Comment:
actually why don't we just have an `append_instruction` method that handles this and that way we don't have to do this here or make fragments public?

Reply:
I need the ID of the last instruction in order to add it to the dependency list and make sure it doesn't accidentally run the httplsn opcode before the `_start` handler is finished, though. (Or we start listening to events before the docker container is built and running for anycloud, for instance.)

Reply:
wouldn't that id be available in the within the handler object the same way you access it here?

Reply:
No. I mean the instruction ID, which will vary from handler to handler and I see no way to get at it after the processing of the handler is complete.

Reply:
Discussed offline, this logic is effectively modifying the agz binary, but it doesn't belong in the compiler and there is no clean way to do it so better to leave it contained in one place. If we find another use case for an internal "event completion" event that we can hang internal code on for something other than this use case (when _start is done, start a web server if some condition is met) we should generalize this.